app/models/thredded/topic.rb

```diff
@@ -15,6 +15,23 @@ class Topic < ActiveRecord::Base
 
     scope :order_sticky_first, -> { order(sticky: :desc) }
 
+    scope :followed_by, lambda { |user|
+      joins('INNER JOIN thredded_user_topic_follows ON thredded_user_topic_follows.topic_id = thredded_topics.id')
+        .where('thredded_user_topic_follows.user_id = ?', user.id)
+    }
+    scope :unread_by, lambda{ |user|
```

Comment:
no I'm being stupid, only unread_by is a duplicate. Have fixed this in the attached PR. If it's good I'll rebase and squash